packages/google_sign_in/google_sign_in_web/lib/src/utils.dart

```diff
@@ -74,10 +77,24 @@ GoogleSignInUserData? gisResponsesToUserData(
     id: payload['sub']! as String,
     displayName: payload['name'] as String?,
     photoUrl: payload['picture'] as String?,
-    idToken: credentialResponse.credential,
+    idToken: credentialResponse!.credential,
```

Comment:
This is conceptually removed enough from the code that ensures that the `!` is safe (it's in another method) that I worry about someone changing that code later and this becoming a crash. Couldn't it just be `credentialResponse?.credential` so it's safe, since it's a nullable field?

Reply:
Yes, in this case `payload != null` implies `credentialResponse != null` but as you said, the connection is pretty vague.

I think what I really want here is an assertion that (credentialResponse?.credential != null) before returning the GoogleSignInUserData, that way I'm expressing in code that the credential must NOT be null before returning a GoogleSignInUserData.